Sources/SwiftToolchain/ToolchainManagement.swift

```diff
@@ -291,4 +284,27 @@ extension FileSystem {
 
     return result.sorted()
   }
+  
+  public func fetchLocalSwiftVersion() throws -> String? {
+    guard let cwd = currentWorkingDirectory else { return nil }
+
+    let versionFile = cwd.appending(component: ".swift-version")
+
+    guard isFile(versionFile), let version = try readFileContents(versionFile)
+      .validDescription?
+      // get the first line of the file
+      .components(separatedBy: CharacterSet.newlines).first,
+      version.contains("wasm")
+    else { return nil }
+    
+    return version
+  }
+  
+  public func setLocalSwiftVersion(_ version: String) throws {
+    guard let cwd = currentWorkingDirectory else { return }
+    
+    let versionFile = cwd.appending(component: ".swift-version")
+    
+    try writeFileContents(versionFile, bytes: ByteString([UInt8](version.utf8)))
```

Comment:
After `versionFilePath` property is created, it can be shorted to this:

```suggestion
    try writeFileContents(versionFilePath, bytes: ByteString([UInt8](version.utf8)))
```